### ISSUE TITLE
Proposal: Allow to return additional information in CompletionItem

### ISSUE THREAD

**Issue by ko1** [microsoft/debug-adapter-protocol#235]
Can we return additional information for users on completion with `CompletionItem`?

Debug adapter can know detailed information of complemented text. For example: On Ruby, if the given text is `1.time`, then it returns `1.times` because DAP can know `1` is `Integer` and it has the method `Integer#times`. On this case, with the completion results `1.times`, showing `Integer#times` is informative for users. In other words, showing type information seems useful.

If the type information is short, it is enough by containing the information in `label`, but it's better to show them separately.

**Comment by connor4312** [comment]
This is requesting the `detail` that is present in the VS Code/LSP CompletionItem:

```
		/**
		 * A human-readable string with additional information
		 * about this item, like type or symbol information.
		 */
		detail?: string;
```

I think copying that property verbatim makes sense for DAP's CompletionItems as well.

**Comment by weinand** [comment]
Yes, a `detail` property on the `completionItem` makes sense.

@connor4312 could you take care of getting this into the spec?

**Comment by weinand** [comment]
@connor4312 thanks for taking care of this.
I've moved the change log item under the previous version because I typically batch all changes under a single version per milestone.
Could you please create an adoption item in VS Code that refers to this item and explains what to do.

**Comment by connor4312** [comment]
Ah, sorry, I didn't realize -- I already copied the change to `vscode-debugadapter-node` and published it as 1.53 following semver. Unpublishing is somewhat dangerous, what should I do to fix this?

**Comment by weinand** [comment]
I've reverted the commit.
